src/platform/apollolake/include/platform/memory.h

```diff
@@ -258,10 +258,16 @@
 #define SRAM_SW_REG_SIZE	0x1000
 
 /* SRAM window 0 FW "registers" */
-#define SRAM_REG_ROM_STATUS	0x0
-#define SRAM_REG_FW_STATUS	0x4
-#define SRAM_REG_FW_TRACEP	0x8
-#define SRAM_REG_FW_END		0xc
+#define SRAM_REG_ROM_STATUS			0x0
+#define SRAM_REG_FW_STATUS			0x4
+#define SRAM_REG_FW_TRACEP			0x8
+#if !CONFIG_DEBUG_IPC_COUNTERS
```

Comment:
This should not be conditional, counters should always be at these offsets even if disabled.

Reply:
these will need to be added to kernel headers too.

Reply:
Updated. I've make PR on the kernel side:
https://github.com/thesofproject/linux/pull/976

Reply:
@lgirdwood What's the point of having them in headers, if these registers are conditional. They can be there, but they don't have to. 3 first registers are used always and logged to the dmesg. These 2 are of new kind that is for debugging very specific purpose (and they may not even be there - depends on kconfig) I'm not sure if kernel should care about them at all. If we want these registers to be added to the kernel code we should log them all the time like I wanted to do in https://github.com/thesofproject/linux/pull/955 . Otherwise as @plbossart mentioned somewhere earlier we will end up with bunch of registers definitions that are not used and people will ask, why they are there at all.

So we should adapt the policy that we log all of them on some kind of related error or we make kernel not care about them and developer will just get what he wants from debugfs.

Reply:
If you make this conditional, then the kernel *needs* a mechanism to know if those counters are present/valid.
The problem with debugfs is that there is no serialization with other events, e.g. you really  want to show those counters when there is an IPC timeout, which you can't do with debugfs.

Reply:
@lgirdwood 
Can't we make those memory windows conditional?